components/shared/code/libs/lib_nvm.c

```diff
@@ -308,19 +380,27 @@ uint32_t lib_nvm_getTotalCycles(void)
 
 void lib_nvm_cleanUp(void)
 {
-    portENTER_CRITICAL();
     data.mcuShuttingDown = true;
+    lib_nvm_check();
+    while(lib_nvm_writesRequired());
```

Comment:
Won't this block and prevent the writes from happening?

Reply:
Yes, which is what you want on cleanup. A calling task shouldnt continue until the nvm writes have all cleaned up, this is used by uds prior to resets